grails-app/controllers/org/pih/warehouse/product/CategoryController.groovy

```diff
@@ -93,11 +101,11 @@ class CategoryController {
     //  @CacheFlush("selectCategoryCache")
     def save() {
         def categoryInstance = new Category(params)
-
-        if (categoryInstance.save(flush: true)) {
+        try {
+            categoryGormService.save(categoryInstance)
             flash.message = "${warehouse.message(code: 'default.created.message', args: [warehouse.message(code: 'category.label', default: 'Category'), categoryInstance.id])}"
             redirect(action: "tree", id: categoryInstance.id)
-        } else {
+        } catch(Exception e) {
```

Comment:
this is not correct. If you get any validation exceptions while saving, you will be redirected to the `tree` anyway and will see the success message, because `save()` in Grails doesn't throw a `ValidationException` when there are any, but returns `null`.

Reply:
Not exactly, using save method provided by `@Service` will throw an exception if it encounters a validation error unlike using `Domain.save()` which will indeed return a null on validation error.

As for using try/catch I believe we talked about letting the generic error handler handle the exception but in the case of  rendering page templates we don't have a generic error handler that can do it and we probably won't have.
I did an investigation into implementing this handler #4240 and unfortunately, I wasn't able to find a proper way to do it.